examples/with-vitest/vite.config.ts

```diff
@@ -1,33 +1,34 @@
 /// <reference types="vitest" />
-/// <reference types="vite/client" />
 
-import { defineConfig } from "vite";
 import solid from "solid-start";
+import { defineConfig, loadEnv } from "vite";
 
-export default defineConfig({
-  test: {
-    globals: true,
-    environment: "jsdom",
-    transformMode: {
-      web: [/\.[tj]sx?$/]
+export default defineConfig(({ mode }) => {
+	// Load env file based on `mode` in the current working directory.
+	// Set the third parameter to '' to load all env regardless of the `VITE_` prefix.
+	const env = loadEnv(mode, process.cwd(), '');
+
```

Comment:
this function stuff is added complexity.. i wonder if we can do without this.. or have the solid-start plugin take care of some of this if its a good idea to do anyway

Reply:
It's the correct way of loading the env-vars in `vite`. We could simplify, but I opted for "correct" over "simple" here as it's an example...

Generally my feeling is that the root-cause should be correctly handled in `solid-testing-library` and it should work in node-envs (which would evaporate the need of most of the setup...)

Reply:
okay makes sense but can we do this also inside a plugin so it doesnt need to be complicated for the user to adopt.. the example is one thing.. but the method should be to incrementally adopt.. so just installing a vite plugin is the easiest thing we can have (and maybe having it as part of solid-start by default)

Reply:
I'm feeling like this is leaving the scope of this pr. It might be worthwhile though to revisit ssr-handling in `vite-plugin-solid` for environments which aren't the browser and have trouble with the "magical" import-logic of `solid/web`.

I would propose to merge this and open another issue/discussion where it feels appropriate to those who are more familiar with the underlying logic than I am atm.